analyzers/src/SonarAnalyzer.Common/Helpers/KnownReference.cs

```diff
@@ -0,0 +1,46 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using static SonarAnalyzer.Helpers.KnownReference.Predicates;
+
+namespace SonarAnalyzer.Helpers
+{
+    public sealed partial class KnownReference
+    {
+        private readonly Func<IEnumerable<AssemblyIdentity>, bool> predicate;
+
+        public static KnownReference XUnit_Assert { get; } = new(
+            NameIs("xunit.assert"),
+            NameIs("xunit").And(VersionLowerThen("2.0")));
+
+        internal KnownReference(Func<AssemblyIdentity, bool> predicate, params Func<AssemblyIdentity, bool>[] or)
```

Comment:
This will simplify it quite a lot. As this is supposed to be instantiated only from within this class (and therefore it would be beneficial to have it `private`), we don't need to enforce it syntactically. You can throw in case it's empty
```suggestion
        internal KnownReference(params Func<AssemblyIdentity, bool>[] orPredicates)
```

Reply:
This would be a bad API design. We can enforce proper usage at compile time and should do so.